skbuild/resources/cmake/UseF2PY.cmake

```diff
@@ -0,0 +1,146 @@
+#
+# ``OUTPUT_VAR <OutputVar>``
+#   Set the variable ``<OutputVar>`` in the parent scope to the path to the
+#   generated source file.  By default, ``<Name>`` is used as the output
+#   variable name.
+#
+# Defined variables:
+#
+# ``<OutputVar>``
+#   The path of the generated source file.
+#
+# Example usage
+# ^^^^^^^^^^^^^
+#
+# .. code-block:: cmake
+#
+#   find_package(F2PY)
+#
+#   # Note: In this case, either one of these arguments may be omitted; their
+#   # value would have been inferred from that of the other.
+#   add_f2py_target(f2py_code f2py_code.pyf)
+#
+#   add_library(f2py_code MODULE ${f2py_code})
+#   target_link_libraries(f2py_code ...)
+#
+#=============================================================================
+# Copyright 2011 Kitware, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+get_property(languages GLOBAL PROPERTY ENABLED_LANGUAGES)
+
+function(add_f2py_target _name)
+  set(options )
+  set(oneValueArgs OUTPUT_VAR)
+  set(multiValueArgs)
+  cmake_parse_arguments(_args "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  list(GET _args_UNPARSED_ARGUMENTS 0 _arg0)
+
+  # if provided, use _arg0 as the input file path
+  if(_arg0)
+    set(_source_file ${_arg0})
+
+  # otherwise, must determine source file from name, or vice versa
+  else()
+    get_filename_component(_name_ext "${_name}" EXT)
+
+    # if extension provided, _name is the source file
+    if(_name_ext)
+      set(_source_file ${_name})
+      string(REGEX REPLACE "\\.[^.]*$" "" _name ${_source})
+
+    # otherwise, assume the source file is ${_name}.pyf
+    else()
+      set(_source_file ${_name}.pyf)
+    endif()
+  endif()
+
+  set(_embed_main FALSE)
+
+  if("C" IN_LIST languages)
+    set(_output_syntax "C")
+  else()
+    message(FATAL_ERROR "C must be enabled to use F2PY")
+  endif()
+
+  set(extension "c")
+
+  set(generated_file "${CMAKE_CURRENT_BINARY_DIR}/${_name}module.${extension}")
+  
+  # The wrappers may but are not always generated. To get around this, we touch
+  set(generated_wrapper "${_name}-f2pywrappers.f")
+  if(NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/${generated_wrapper}")
+    file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/${generated_wrapper}" "")
+  endif()
+
+  get_filename_component(generated_file_dir ${generated_file} DIRECTORY)
+  
+  # FIXME: possible bug
+  get_filename_component(f2py_module_name generated_file NAME_WE)
```

Comment:
The reason that I'm hesitant to change this is that the module flag isn't used in SciPy and compiler failures were much more explicit than hunting down an obscure bug.